source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,666 @@
+SDK_ALIGN( static rx_buffer_t rxDataBuff_0[ ENET_RXBUFF_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+SDK_ALIGN( static tx_buffer_t txDataBuff_0[ ENET_TXBD_NUM ], FSL_ENET_BUFF_ALIGNMENT );
+
+static void prvEMACHandlerTask( void * pvParameters );
+
+static void ethernet_callback( ENET_Type * base,
+                               enet_handle_t * handle,
+                               enet_event_t event,
+                               enet_frame_info_t * frameInfo,
+                               void * userData );
+
+static void prvProcessFrame( int length );
+
+BaseType_t xNetworkInterfaceInitialise( void )
+{
+    status_t xStatus;
+    phy_config_t xConfig = { 0 };
+    BaseType_t xResult = pdFAIL;
+    phy_speed_t speed;
+    phy_duplex_t duplex;
+    status_t readStatus;
+    enet_config_t config;
+    uint32_t sysClock;
+    enet_buffer_config_t buffCfg[ ENET_RING_NUM ];
+    int i;
+    uint8_t ucCounter = 0;
+    bool autoNegotiationComplete = false;
+    bool LinkUp = false;
+
+    configASSERT( FSL_FEATURE_ENET_QUEUE == 1 );
+
+    #if defined( ENET_ENHANCEDBUFFERDESCRIPTOR_MODE )
+        configASSERT( pdFALSE );
+    #endif
+
+    switch( eEMACState )
+    {
+        case xEMAC_SetupPHY:
+            ethernetifLocal->RxBuffDescrip = &( rxBuffDescrip_0[ 0 ] );
+            ethernetifLocal->TxBuffDescrip = &( txBuffDescrip_0[ 0 ] );
+            ethernetifLocal->RxDataBuff = &( rxDataBuff_0[ 0 ] );
+            ethernetifLocal->TxDataBuff = &( txDataBuff_0[ 0 ] );
+            ethernetifLocal->base = ( void * ) ENET_BASE;
+
+            /* Set the clock frequency. */
+            mdioHandle.resource.csrClock_Hz = EXAMPLE_CLOCK_FREQ;
+            mdioHandle.resource.base = ( void * ) ENET_BASE;
+
+            /* Set the configuration to auto-negotiate; set the phy to full duplex mode; the phy's address; and the speed. */
+            xConfig.autoNeg = pdTRUE;
+            xConfig.duplex = kPHY_FullDuplex;
+            xConfig.phyAddr = BOARD_ENET0_PHY_ADDRESS;
+            xConfig.speed = kPHY_Speed100M;
+
+            FreeRTOS_printf( ( "Starting PHY initialization.\r\n" ) );
+
+            xStatus = PHY_Init( &phyHandle, &xConfig );
+
+            if( xStatus != kStatus_Success )
+            {
+                FreeRTOS_printf( ( "Failed to initialize the PHY..." ) );
+                break;
+            }
+
+            eEMACState = xEMAC_WaitPHY;
+        /* Fall through. */
+
+        case xEMAC_WaitPHY:
+            FreeRTOS_printf( ( "Configuration successful. Waiting for auto-negotiation to complete...\r\n" ) );
+
+            do
+            {
+                readStatus = PHY_GetAutoNegotiationStatus( &phyHandle, &autoNegotiationComplete );
+
+                if( autoNegotiationComplete == true )
+                {
+                    break;
+                }
+
+                /* Try for only a limited number of times. */
+                if( ucCounter++ > MAX_AUTONEG_FAILURE_COUNT )
```

Comment:
`MAX_AUTONEG_FAILURE_COUNT = 5`
When you find the time it would be interesting to see how many loops are actually needed.

Reply:
I increased this to 40 and the timeout to 100ms. It seems that 4s is the cap on the time required to complete auto-negotiation in many stacks. I followed suit there.